invokeai/app/services/bulk_download/bulk_download_base.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+from invokeai.app.services.invoker import Invoker
+
+
+class BulkDownloadBase(ABC):
+    @abstractmethod
+    def start(self, invoker: Invoker) -> None:
+        """
+        Starts the BulkDownloadService.
+
+        This method is responsible for starting the BulkDownloadService and performing any necessary initialization
+        operations to prepare the service for use.
+
+        param: invoker: The Invoker that holds all the services, required to be passed as a parameter to avoid circular dependencies.
+        """
+
+    @abstractmethod
+    def __init__(self, output_folder: Union[str, Path]):
+        """
+        Create BulkDownloadBase object.
+
+        :param output_folder: The path to the output folder where the bulk download files can be temporarily stored.
```

Comment:
Why do we need to provide a real folder path?

Reply:
I was thinking that this was necessary because of folder permissions of the app itself. I can test to see if this is necessary.

I also think the added benefit of the state where the app is not shut down properly and the bulk_download folder persists it will be close to the rest of the invoke stuff.